src/app/shell/admin-tools/platform/create-direction/directions/directions.component.ts

```diff
@@ -57,6 +63,14 @@ export class DirectionsComponent implements OnInit, OnDestroy {
         }
       });
 
+  this.directionsPerPage$ === null;
+  this.directionsPerPage$
+    .pipe(
+      takeUntil(this.destroy$)
+      ).subscribe((directionsPerPage: number)=>{
+      this.directionsPerPage = directionsPerPage;
+      this.store.dispatch(new GetFilteredDirections());
```

Comment:
move this.store.dispatch(new GetFilteredDirections()); out of the subscribtion and remove subscrbtion. pass value via async in template